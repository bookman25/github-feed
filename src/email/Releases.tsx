import * as React from 'react';

export interface IReleases {
	releases: IRelease[];
}

export interface IRelease {
	html_url?: string;
	name?: string;
	tag_name?: string;
}

export function Releases({ releases }: IReleases) {
	if (!releases || releases.length === 0) {
		return null;
	}

	const release = releases[0];
	return (
		<p>
			<strong>Latest release: </strong>
			<a href={ release.html_url }>{ release.name || release.tag_name }</a>
		</p>
	);
}