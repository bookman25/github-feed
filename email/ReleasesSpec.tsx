jest.unmock('./Releases');

import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Releases from './Releases';

describe('Releases', () => {
	it('with no releases', () => {
		const view = shallow(<Releases releases={ [] } />);
		expect(view.type()).toBeNull();
	});

	it('with a release name', () => {
		const view = shallow(<Releases releases={ [{ name: 'name', html_url: 'url' }] } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});
	it('without a release name and only a tag', () => {
		const view = shallow(<Releases releases={ [{ tag_name: 'name', html_url: 'url' }] } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});
});