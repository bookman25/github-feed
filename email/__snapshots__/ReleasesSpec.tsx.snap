exports[`Releases with a release name 1`] = `
<p>
  <strong>
    Latest release: 
  </strong>
  <a
    href="url">
    name
  </a>
</p>
`;

exports[`Releases without a release name and only a tag 1`] = `
<p>
  <strong>
    Latest release: 
  </strong>
  <a
    href="url">
    name
  </a>
</p>
`;
